app/ui/legacy/src/main/java/com/fsck/k9/activity/MessageList.kt

```diff
@@ -879,9 +879,11 @@ open class MessageList :
         } else if (id == R.id.move_to_drafts) {
             messageViewFragment!!.onMoveToDrafts()
             return true
-        } else if (id == R.id.show_headers || id == R.id.hide_headers) {
-            messageViewFragment!!.onToggleAllHeadersView()
-            updateMenu()
+        } else if (id == R.id.show_headers) {
+            val intent = Intent(this, MessageHeadersActivity::class.java)
+            val reference = messageViewFragment!!.messageReference.toIdentityString()
+            intent.putExtra(MessageHeadersActivity.ARG_REFERENCE, reference)
```

Comment:
This should be moved to `MessageHeadersActivity.kt`. Maybe as `createLaunchIntent()`.